csharp/src/Apache.Arrow.Flight/Middleware/MetadataAdapter.cs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Apache.Arrow.Flight.Middleware.Interfaces;
+using Grpc.Core;
+
+namespace Apache.Arrow.Flight.Middleware;
+
+public class MetadataAdapter : ICallHeaders
```

Comment:
This class doesn't appear to be used in this PR. Is it possible it's been superseded by `CallHeaders`?